### PR TITLE
Error-check split, simplify code

### LINKED ISSUE #9404: Splits do not check if the variables passed in exist
### Description of the enhancement or error report
Splits do not check if the variables passed in exist. If you pass in a non-existing variable name you get a cryptic PETSc error (cryptic for the average user).

### Rationale for the enhancement or information for reproducing the error
We can do better. I'm adding a quick check that errors out if the variable does not exist in teh problem

### Identified impact
Improved input checking

### PR DESCRIPTION
**Pull request by dschwen**
Closes #9404

### CODE REVIEW

**Comment by dschwen** [comment]
Btw. this adds a new `Moose::stringify` specialization

```
template <typename T>
std::string
stringify(const std::vector<T> & v)
```

Which allows me to save quite a bit of code in `Split`. It generates a comma separated list of stringified vector entries. Might be useful in other places as well.

File: framework/include/utils/Conversion.h [hunk 1]
```diff
@@ -75,6 +75,21 @@ std::string stringify(const SolveType & t);
 template <>
 std::string stringify(const ExecFlagType & t);
 
+/// Add no-op stringify if the argument already is a string
+template <>
+std::string stringify(const std::string & t);
```
Comment (permcody) [inline framework/include/utils/Conversion.h]: We have an "infix iterator" header: https://github.com/idaholab/moose/blob/devel/framework/include/utils/InfixIterator.h that allows you to do this kind of thing in a very generic context. @aeslaughter was also trying to add a `join()` method to MooseUtils, but with the iterator and the STL you can insert commas between items in any context pretty easily. 

We certainly don't need two (or any 😉).
https://github.com/idaholab/moose/pull/9242/files#diff-b667c5b178211dc59f0967396aeaf061R399

Comment (dschwen) [inline framework/include/utils/Conversion.h]: `infix_iterator` is not used in MOOSE. I suggest we just delete that one.

Comment (permcody) [inline framework/include/utils/Conversion.h]: That's true, but `infix_ostream_iterator` is used all over the place!

**Review by permcody — CHANGES_REQUESTED** [review]
Per our discussion, we'd like a more general capability for this operation since we already have one. I should take as input the delimiter and work with more containers.

**Comment by dschwen** [comment]
Have you seen the unit test?!

**Review by permcody — APPROVED** [review]
👍

**Pull request merged**